core/src/main/java/discord4j/core/object/entity/Member.java

```diff
@@ -368,6 +363,32 @@ public String getNicknameMention() {
         return getClient().getMemberById(getGuildId(), id).flatMap(this::isHigher);
     }
 
+    /**
+     * Requests to determine if the position of this member's highest role is greater than the highest position of the
+     * provided roles or signal IllegalArgumentException if the provided roles are from a different guild than this
+     * member.
+     *
+     * @param otherRoles The list of roles to compare in the role hierarchy with this member's roles.
+     * @return A {@link Mono} where, upon successful completion, emits {@code true} if the position of this member's
+     * highest role is greater than the highest position of the provided roles, {@code false} otherwise.
+     * If an error is received it is emitted through the {@code Mono}.
+     */
+    public Mono<Boolean> hasHigherRoles(List<Role> otherRoles) {
```

Comment:
I don't know about that. It seems to me that the best type would actually be `Collection`. Nowhere are any `List` or `Set` exclusive methods used, nor are there any indexed lookups (for which a `Set` would be superior), only iteration.

Reply:
It's not logical for there to be duplicate roles.

Reply:
It's not, but having duplicates wouldn't affect the functionality here.

Reply:
I figure `List` was chosen to be friends with `Flux#collectList` that is the shortest way to turn a sequence into an `Iterable`, which should be the type used here.

Reply:
What should I do then ? I used List because Flux.collectList() is, for, me, the most common method like quantic said but Flux.collect(Collectives.toSet()) can also be used

Reply:
Use `Iterable` like @quanticc said, that's the widest type.